### DISCUSSION TITLE
v7.5.1

### DISCUSSION THREAD

**Discussion by ssorallen** [tabwrangler/tabwrangler#471] [General]
Maintain locks and tab times when browser "discards" tabs. When tabs are "discarded" to free memory and system resources the tabs' IDs change - Tab Wrangler was not correctly tracking those ID updates and was losing state.

Issues

* [#468] Timer is reset after hiberante/sleep
* [#462] It will auto unlock the locked tab

<hr /><em>This discussion was created from the release <a href='https://github.com/tabwrangler/tabwrangler/releases/tag/v7.5.1'>v7.5.1</a>.</em>

**Comment by ssorallen** [comment]
v7.5.1 was uploaded to the Chrome Webstore at the time of this comment.

**Comment by ssorallen** [comment]
✅ v7.5.1 is live in the Chrome Webstore as of the time of this comment.

**Comment by adacole7** [comment]
After this latest update, my wrangler is still not closing tabs.  Last tabs were closed 1 month ago.  Any suggestions?

**Comment by ssorallen** [comment]
@adacole7 Could you please check Tab Wrangler's logs and paste them here?

1. Go to _Window_ > _Extensions_
2. Toggle "Developer Mode" in upper right corner
3. Click the "service worker" link that says Inspect views service worker next to Tab Wrangler
4. Copy/paste the contents of the "Console" tab here.

**Comment by TheIndieArmy** [comment]
Even on this version I still have issue where tabs will auto-unlock themselves. Still seems to only be occurring when system RAM usage is high (~95%). 

Not sure if there is anything helpful in this log (no errors were present).

> [menus] Updating context menu for tab ID  2025593438
> background.entry.js:1035 [onNewTab] updating new tab Object
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 159.437s ago
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593438
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 34.985s ago
> background.entry.js:1035 [onNewTab] updating new tab Object
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 99.997s ago
> background.entry.js:1035 [onNewTab] updating new tab Object
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593688
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 35.004s ago
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 100.015s ago
> background.entry.js:1035 [onNewTab] updating new tab Object
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 6.464s ago
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 71.474s ago
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593688
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 136.482s ago
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 26.548s ago
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593438
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 91.56s ago
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593441
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025592441
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593438
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 156.564s ago
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 35.005s ago
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593720
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593688
> background.entry.js:1035 [onNewTab] updating new tab Object
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 100.011s ago
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025592355
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 35.003s ago
> background.entry.js:1035 [onNewTab] updating new tab Object
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025592355
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 100.007s ago
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 35.007s ago
> background.entry.js:1035 [onNewTab] updating new tab Object
> background.entry.js:1035 [onNewTab] updating new tab Object
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 100.015s ago
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 11.608s ago
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025592355
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593731
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593728
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025592355
> background.entry.js:1035 [onNewTab] updating new tab Object
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 76.619s ago
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025592355
> background.entry.js:1035 [onNewTab] updating new tab Object
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 141.628s ago
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025592355
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 23.59s ago
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025592355
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593741
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 88.592s ago
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025592355
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593740
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593728
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025592355
> background.entry.js:1035 [onNewTab] updating new tab Object
> background.entry.js:1035 [onNewTab] updating new tab Object
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 153.603s ago
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 35.004s ago
> background.entry.js:1035 [onNewTab] updating new tab Object
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 100.02s ago
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593441
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 34.995s ago
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593438
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593728
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593438
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593749
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593438
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 99.999s ago
> background.entry.js:1035 [onNewTab] updating new tab Object
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 34.996s ago
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 100.002s ago
> background.entry.js:1035 [onNewTab] updating new tab Object
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593770
> background.entry.js:1035 [onNewTab] updating new tab Object
> background.entry.js:1035 [onNewTab] updating new tab Object
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593773
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593776
> background.entry.js:1035 [onNewTab] updating new tab Object
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 34.999s ago
> background.entry.js:1035 [onNewTab] updating new tab Object
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593782
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593779
> background.entry.js:1035 [onNewTab] updating new tab Object
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593779
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 100.013s ago
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593784
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593779
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593784
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 21.755s ago
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593779
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 86.764s ago
> background.entry.js:1798 [lostEventsWatchdog]: last alarm 151.772s ago
> background.entry.js:1521 [menus] Updating context menu for tab ID  2025593784

**Comment by ssorallen** [comment]
@TheIndieArmy Thank you for the report. I will copy your logs into #474 and track down what is happening.

I am using v7.5.1 as well, and Tab Wrangler is working as expected for me. I need to figure out how to reproduce this issue.

**Comment by ssorallen** [comment]
v7.6.0 is now the latest release: https://github.com/tabwrangler/tabwrangler/discussions/485
